app/Http/Requests/BulkUserCreateRequest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class BulkUserCreateRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $circle_id = $this->route('circle_id');
+
+        return [
+            'users' => ['required', 'array', 'min:1'],
+            'users.*.address' => ['required', 'string', 'size:42', 'distinct',
+                // validate id is unique within circle
+                Rule::unique('users')->where(function ($query) use ($circle_id) {
+                    return $query->where('circle_id', $circle_id)->whereNull('deleted_at');
```

Comment:
Does Laravel have support for default scopes or some way that we can more easily verify " deleted_at is NULL" all the time?

Reply:
added [withoutThrashed](https://github.com/laravel/framework/pull/38314)